source/time_stepping/convection_time_step.cc

```diff
@@ -51,11 +57,36 @@ namespace aspect
             fe_values.reinit (cell);
             fe_values[this->introspection().extractors.velocities].get_function_values (this->get_solution(),
                                                                                         velocity_values);
+            in.reinit(fe_values, cell, this->introspection(), this->get_solution());
+            this->get_material_model().fill_additional_material_model_inputs(in, this->get_solution(), fe_values, this->introspection());
+            this->get_material_model().evaluate(in, out);
 
             double max_local_velocity = 0;
             for (unsigned int q=0; q<n_q_points; ++q)
-              max_local_velocity = std::max (max_local_velocity,
-                                             velocity_values[q].norm());
+              {
+                if (this->introspection().compositional_name_exists("porosity"))
```

Comment:
Can you make this condition dependent on the field type (`this->introspection().composition_type_exists(CompositionalFieldDescription::porosity)`)? Also please move this to the top of the file:
```
const bool consider_darcy_timestep = ...
```